pithos/plugins/notification_icon.py

```diff
@@ -91,7 +111,18 @@ def on_has_name_owner(bus, result):
                 self.prepare_complete()
             else:
                 # No fancy tray here
-                self.prepare_complete()
+                logging.info('Checking if org.gnome.Shell is owned and version')
+                Gio.DBusProxy.new(
+                    self.bus,
+                    Gio.DBusProxyFlags.NONE,
+                    None,
+                    'org.gnome.Shell',
```

Comment:
Looks like you got some work to do then ;P

In my mind you'd want to be able to lock it down to the point that you could even specify if the app is even allowed to set read/write props.